### PR TITLE
refactor(textarea): apply common form logic

### PR DESCRIPTION
**Pull request by dpellier**

### CODE REVIEW

**Review by aesteves60 — APPROVED** [review]

File: packages/components/src/textarea/src/components/osds-textarea/constants/default-attributes.ts
Comment (astagnol) [inline packages/components/src/textarea/src/components/osds-textarea/constants/default-attributes.ts:1]: cols and rows default attributes no set?

**Review by astagnol — CHANGES_REQUESTED** [review]
- inline, resizable, cols and rows controls in Storybook are not working
- contrasted control should be removed from Storybook

File: packages/components/src/textarea/src/components/osds-textarea/constants/default-attributes.ts
Comment (dpellier) [reply packages/components/src/textarea/src/components/osds-textarea/constants/default-attributes.ts:1]: They're optional, so unless we actually want a real default value, there is no need to set them.
And for what I see in the existing component, they were set a `undefined` by default

File: packages/components/src/textarea/src/components/osds-textarea/osds-textarea.tsx [hunk 1]
```diff
@@ -1,289 +1,136 @@
...
     return (
-      <Host {...{
-        class: {
-          'ods-error': Boolean(this.controller.hasError()),
-        },
-        tabindex: textInputTabIndex,
-        hasFocus,
-      }}
-      >
+      <Host class={{ 'ods-error': this.internalError }}>
```
Comment (aesteves60) [inline packages/components/src/textarea/src/components/osds-textarea/osds-textarea.tsx:113]: I think we need the tabindex here or the component will not be reflect on the host

File: packages/storybook/stories/controls.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  CONTROL_CATEGORY.accessibility,
+];
+
+function orderControls(control: Record<string, any>) {
+  try {
+    return Object.entries(control)
+      .sort((a, b) => {
+        return orderedControlCategories.indexOf(a[ 1 ].category) - orderedControlCategories.indexOf(b[ 1 ].category);
+      })
+      .reduce((res, entry) => {
```
Comment (aesteves60) [inline packages/storybook/stories/controls.ts:21]:
```suggestion
      .reduce((res, { key, value }) => {
```

**Review by aesteves60 — COMMENTED** [review]
You need to run yarn doc & remove the unused props like prefixValue or step

File: packages/storybook/stories/controls.ts (hunk 2 above)
Comment (dpellier) [reply packages/storybook/stories/controls.ts:21]: entry is an array containing [<prop>, <value>] not an object

**Review by aesteves60 — APPROVED** [review]

**Review by astagnol — APPROVED** [review]

**Pull request merged**